test.js

```diff
@@ -12,3 +12,63 @@ test('sync', t => {
 	t.is(unusedFilename.sync('fixtures/unicorn.txt'), 'fixtures/unicorn (1).txt');
 	t.is(unusedFilename.sync('fixtures/rainbow.txt'), 'fixtures/rainbow (3).txt');
 });
+
+test('options.incrementer', async t => {
+	const testFile = 'fixtures/rainbow.txt';
+
+	const incrementerTest = async ({incrementer, error, message, result}) => {
+		if (error) {
+			t.throws(
+				() => unusedFilename.sync(testFile, {incrementer}),
+				error,
+				message
+			);
+
+			await t.throwsAsync(
+				async () => unusedFilename(testFile, {incrementer}),
+				error,
+				message
+			);
+		} else {
+			t.is(
+				unusedFilename.sync(testFile, {incrementer}),
+				result,
+				message
+			);
+
+			t.is(
+				await unusedFilename(testFile, {incrementer}),
+				result,
+				message
+			);
+		}
+	};
+
+	await Promise.all(
+		[
+			{
+				incrementer: '',
+				error: TypeError,
+				message: '`incrementer` should be a function'
+			},
+			{
+				incrementer() {},
+				error: TypeError,
+				message: '`incrementer` should return a string'
+			},
+			{
+				incrementer() {
+					return 'rainbow.txt';
+				},
+				error: Error,
+				message: '`incrementer` should return unique names'
+			},
+			{
+				incrementer(filename, extension, counter) {
+					return `${filename}${' - copy'.repeat(counter)}${extension}`;
+				},
+				result: 'fixtures/rainbow - copy - copy - copy.txt'
+			}
+		].map(incrementerTest)
```

Comment:
Instead of mapping here, use test macros: https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#reusing-test-logic-through-macros

Reply:
separate test?

```text
  √ sync
  √ `incrementer` should be a function
  √ `incrementer` should return a string
  √ `incrementer` should return unique names
  √ custom incrementer
  √ async
```

Reply:
Yes